src/prisma/schema.prisma

```diff
@@ -103,6 +103,7 @@ model WalletNonce {
 
 model Transactions {
   id                        String    @id @default(uuid()) @map("id")
+  idempotencyKey            String    @unique @default(uuid()) @map("idempotencyKey")
```

Comment:
- `@unique` - Creates a unique index for O(1) lookups. Ensures exactly one tx for a given idempotency key.
- `@default(uuid())` - For backward compatibility. When introducing this change to a new DB, there needs to be non-null values to satisfy the uniqueness constraint.